src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,965 @@
+
+    def _setScanDelay(self, scanDelay):
+        """
+        Setter which checks if detector can record images before ebeam scanner has started to scan.
+
+        :param pixelSize (tuple):
+        :return (tuple):
+        """
+        # Check if detector can record images before ebeam scanner has started to scan.
+        if not (hasattr(self.parent, "_mppc")) or self.parent._mppc.acqDelay.value >= scanDelay[0]:
+            return scanDelay
+        else:
+            # Change values so that 'self.parent._mppc. acqDelay.value - self.scanDelay.value[0]' has a positive result
+            logging.warning("Detector cannot record images before ebeam scanner has started to scan.\n"
+                            "Detector needs to start after scanner.")
+            logging.info("The entered acquisition delay is %s in the eBeamScanner and the scan delay in the MPPC is "
+                         "%s" % (self.parent._mppc.acqDelay.value, scanDelay[0]))
+            return self.scanDelay.value
+
+
+class MirrorDescanner(model.Emitter):
+    """
+    Represents the Mirror de scanner of a single field image which counter scans the scanner movement to ensure that
+    the collected signal always hits the center of each mppc cell on the detector.
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Initializes Mirror de scanner of a single field image which counter scans the scanner movement to ensure
+        that the collected signal always hits the center of each mppc cell on the detector.
+
+        :param name:
+        :param role:
+        :param parent:
+        :param kwargs:
+        """
+        super(MirrorDescanner, self).__init__(name, role, parent=parent, **kwargs)
+
+        self.rotation = model.FloatContinuous(0, range=(0, 2 * numpy.pi), unit='rad')
+        self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V')
+        self.scanGain = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V')
+
+
+class MPPC(model.Detector):
+    """
+    Represents the camera (mppc sensor) for acquiring the image data.
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Initializes the camera (mppc sensor) for acquiring the image data.
+
+        :param name:
+        :param role:
+        :param parent:
+        :param kwargs:
+        """
+        super(MPPC, self).__init__(name, role, parent=parent, **kwargs)
+
+        self._server_URL = self.parent._server_url
+
+        self._shape = (8, 8)
+        self.filename = model.StringVA(time.strftime("default--%Y-%m-%d-%H-%M-%S"), setter=self._setFilename)
+        self.dataContent = model.StringEnumerated('empty', DATA_CONTENT_TO_ASM.keys())
+        # NOTE: Do not write real username/password here since this is published on github in plaintext!
+        self.externalStorageURL = model.VigilantAttribute(urlparse('ftp://username:password@example.com/Pictures'),
```

Comment:
If you need to enter the path, username, or password separately then you need to convert it via urlparse.
Currently I always use parts of the full address instead of the full address. So the urlparse object makes a lot of sense to me. But maybe I miss something? (if one likes to use the entire address one can now use urlunparse, but that is a bit of a work around) 
>Note this part is moved to the AcquisitionServer class

Reply:
`urlparse` returns a special object. The content of the VA should be the simplest format -> a string. Otherwise you cannot set it from a YAML file or from the CLI.

Reply:
Aaah I didn't know that. Then it does make a lot of sense to indeed not use urlparse. I will acces this string it in another way (via a getter or just directly transforming it before calling)